dnplab/io/mat.py

```diff
@@ -0,0 +1,150 @@
+"""
+Save or load mat file 
+
+Author: Yen-Chun Huang, Timothy Keller
+
+"""
+
+from scipy.io import savemat, loadmat
+import numpy as np
+from numpy import array, uint32
+import os
+import dnplab as _dnp
+
+
+def save_mat(data, path, **kwargs):
+    """Save DNPLab data object as Matlab .mat file
+
+    Args:
+        data (DNPData): Dictionary of data
+        filename (str): directory to save file
+
+    """
+
+    matlab_dict = {}
+
+    matlab_dict["values"] = data.values
+    matlab_dict["dims"] = data.dims
+    matlab_dict["coords"] = data.coords.coords
+    matlab_dict["attrs"] = str(data.attrs)
+    matlab_dict["dnplab_attrs"] = str(data.dnplab_attrs)
+    matlab_dict["proc_attrs"] = str(data.proc_attrs).replace(
+        "None", "'__PYTHON_NONE__'"
+    )
+
+    savemat(path, matlab_dict, **kwargs)
+
+
+def import_mat(path):
+    """
+    Import mat file and return DNPData object
+
+    Args:
+        path (str):                 Path to .mat file
+
+    Returns:
+        data (DNPData):             DNPData objects
+    """
+
+    if path[-1] == os.sep:
+        path = path[:-1]
+
+    if ".mat" not in path:
+        raise TypeError("Incorrect file type, must be .mat")
+
+    data = loadmat("test.mat")
+    values = _get_values(data)
+    shape = np.shape(values)
+    dims = _get_dims(data, shape)
+    coords = _get_coords(data, shape)
+    attrs = _get_attrs(data, "attrs")
+    dnplab_attrs = _get_attrs(data, "dnplab_attrs")
+    proc_attrs = eval(
+        str(_get_attrs(data, "proc_attrs")).replace("__PYTHON_NONE__", "None")
+    )
```

Comment:
This is very insecure. We should remove this feature until we have a better way to implement loading in attributes.